cdap-master/src/main/java/co/cask/cdap/data/tools/ReplicationStatusTool.java

```diff
@@ -0,0 +1,379 @@
+import co.cask.cdap.proto.id.NamespaceId;
+import co.cask.cdap.replication.ReplicationConstants;
+import com.google.common.base.Throwables;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.Cell;
+import org.apache.hadoop.hbase.CellUtil;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.Result;
+import org.apache.hadoop.hbase.client.ResultScanner;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.StringTokenizer;
+
+/**
+ * The Replication Status tool is used to track the status of replication from Master to a Slave Cluster.
+ * The tool reads writeTime and replicateTime for all tables and uses it to determine if all CDAP tables have been
+ * completely replicated to the Slave cluster.
+ * */
+public class ReplicationStatusTool {
+
+  private static CConfiguration cConf = CConfiguration.create();
+  private static Configuration hConf = HBaseConfiguration.create();
+  protected static final Logger LOG = LoggerFactory.getLogger(ReplicationStatusTool.class);
+
+  private static String masterClusterID = null;
+  private static String slaveClusterID = null;
+  private static String localClusterID = null;
+  private static String masterChecksum = null;
+  private static Long masterShutdownTime = 0L;
+  private static String inputStatusFileName = null;
+  private static String outputStatusFileName = null;
+  private static boolean debugDump = false;
+  private static String delimiter = ":";
+
+
+  public static void main (String args[]) throws Exception {
+
+    parseArgs(args);
+    localClusterID = hConf.get(ReplicationConstants.ReplicationStatusTool.CLUSTER_ID);
+    if (localClusterID == null) {
+      System.out.println("Can't read "
+                           + ReplicationConstants.ReplicationStatusTool.CLUSTER_ID + "from Configuration.");
+    } else {
+      LOG.info("Local clusterID from Configuration = " + localClusterID);
+    }
+
+    if (debugDump && masterClusterID != null) {
+      dumpReplicationStateTable();
+      return;
+    }
+
+    if (masterClusterID == null
+      || (localClusterID.matches(masterClusterID) && outputStatusFileName == null)
+      || (!localClusterID.matches(masterClusterID) && inputStatusFileName == null)) {
+      showHelp();
+      return;
+    } else if (localClusterID.matches(masterClusterID)) {
+      processMasterCluster();
+    } else {
+      processSlaveCluster();
+    }
+  }
+
+  private static void parseArgs(String args[]) {
+    for (int i = 0; i < args.length;) {
+      if (args[i].charAt(0) == '-') {
```

Comment:
There are libraries that help with command line options instead of parsing everything yourself. You can take a look at how CLIMain or co.cask.cdap.etl.tool.UpgradeTool uses Options. Though I think we can do that in a follow up PR as well.